core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java

```diff
@@ -876,6 +879,196 @@ public void testNaturalJoinNotSupported()
                 .hasErrorCode(NOT_SUPPORTED);
     }
 
+    @Test
+    public void testWindowClause()
+    {
+        assertFails("SELECT * FROM t1 WINDOW w AS (PARTITION BY a), w AS (PARTITION BY a)")
+                .hasErrorCode(DUPLICATE_WINDOW_NAME);
+
+        assertFails("SELECT * FROM t1 WINDOW w AS (PARTITION BY a), w AS (ORDER BY b)")
+                .hasErrorCode(DUPLICATE_WINDOW_NAME);
+
+        assertFails("SELECT * FROM t1 WINDOW w AS (), w1 as (), w AS (w)")
+                .hasErrorCode(DUPLICATE_WINDOW_NAME);
+
+        // window definitions with the same names are allowed in different query specifications (in this case, outer and inner query)
+        analyze("SELECT * FROM (SELECT * FROM t1 WINDOW w AS (PARTITION BY a)) " +
+                "WINDOW w AS (PARTITION BY a)");
+
+        // window definition in inner query is not visible in outer query
```

Comment:
Add a test for the opposite case: a window in the outer query cannot be referenced by the inner query.